### PR TITLE
Error message improvement. show "Cannot find namespace '{0}'." message.

### PR DESCRIPTION
**Pull request by vvakame**
implement #722

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -801,7 +801,13 @@ module ts {
 
             let symbol: Symbol;
             if (name.kind === SyntaxKind.Identifier) {
-                symbol = resolveName(name, (<Identifier>name).text, meaning, Diagnostics.Cannot_find_name_0, <Identifier>name);
+                let message: DiagnosticMessage;
+                if (meaning === SymbolFlags.Namespace) {
```
Comment (mhegazy) [inline src/compiler/checker.ts]: what about type?

Comment (vvakame) [inline src/compiler/checker.ts]: type? it means #102?

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -801,7 +801,13 @@ module ts {
 
             let symbol: Symbol;
             if (name.kind === SyntaxKind.Identifier) {
-                symbol = resolveName(name, (<Identifier>name).text, meaning, Diagnostics.Cannot_find_name_0, <Identifier>name);
+                let message: DiagnosticMessage;
+                if (meaning === SymbolFlags.Namespace) {
+                    message = Diagnostics.Cannot_find_namespace_0;
+                } else {
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: `else` on the next line

File: src/compiler/checker.ts [hunk 3]
```diff
@@ -801,7 +801,13 @@ module ts {
...
             let symbol: Symbol;
             if (name.kind === SyntaxKind.Identifier) {
-                symbol = resolveName(name, (<Identifier>name).text, meaning, Diagnostics.Cannot_find_name_0, <Identifier>name);
+                let message: DiagnosticMessage;
+                if (meaning === SymbolFlags.Namespace) {
+                    message = Diagnostics.Cannot_find_namespace_0;
+                } else {
+                    message = Diagnostics.Cannot_find_name_0;
+                }
+                symbol = resolveName(name, (<Identifier>name).text, meaning, message, <Identifier>name);
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: Add a space above

File: src/compiler/checker.ts [hunk 4]
```diff
@@ -801,7 +801,13 @@ module ts {
 
             let symbol: Symbol;
             if (name.kind === SyntaxKind.Identifier) {
-                symbol = resolveName(name, (<Identifier>name).text, meaning, Diagnostics.Cannot_find_name_0, <Identifier>name);
+                let message: DiagnosticMessage;
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: Use a ternary expression (`?`/`:`

File: src/compiler/checker.ts (hunk 1 above)
Comment (mhegazy) [inline src/compiler/checker.ts]: yeah.. we can do it later on though

File: src/compiler/checker.ts (hunk 2 above)
Comment (JsonFreeman) [inline src/compiler/checker.ts]: Put open brace on next line

File: src/compiler/checker.ts (hunk 4 above)
Comment (JsonFreeman) [inline src/compiler/checker.ts]: I agree, I like ternary better here

**Comment by JsonFreeman** [comment]
:+1:

**Comment by mhegazy** [comment]
@vvakame can you refresh the PR. i think we are good to go.

File: src/compiler/checker.ts (hunk 1 above)
Comment (vvakame) [inline src/compiler/checker.ts]: sorry, I tried it an hour, but I can't defeat it...

**Comment by vvakame** [comment]
thank you for reviewing! :smile_cat:

**Comment by mhegazy** [comment]
Thanks again @vvakame. manually merged. there was an issue of baselines including tests\baselines\references\ProjectOutput. this folder is just a temp folder and should not be included in the test.

Mocha should be cleaning up the folder after a test run. but i believe on node 0.10.* there is an issue that causes it to just fail on the first exception, and do not run the cleanup. @danquirk can you checkin your script in the script directory so that everyone can run it, since the issue is not going to be fixed any time soon.

**Comment by vvakame** [comment]
yay! :smile_cat: 

FYI. 

```
$ node -v
v2.0.0
$ uname -a
Darwin katharina.local 14.3.0 Darwin Kernel Version 14.3.0: Mon Mar 23 11:59:05 PDT 2015; root:xnu-2782.20.48~5/RELEASE_X86_64 x86_64
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.10.3
BuildVersion:	14D136
$ cat .git/config
cat .git/config
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	autocrlf = input
	whitespace = cr-at-eol
[remote "origin"]
	url = git@github.com:Microsoft/TypeScript.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[remote "vvakame"]
	url = git@github.com:vvakame/TypeScript.git
	fetch = +refs/heads/*:refs/remotes/vvakame/*
[branch "typeguardsByConstructorSigniture"]
	remote = vvakame
	merge = refs/heads/typeguardsByConstructorSigniture
[branch "improveModuleMissingErrorMessage"]
	remote = vvakame
	merge = refs/heads/improveModuleMissingErrorMessage
```

**Pull request closed without merging**